cohortextractor2/dsl.py

```diff
@@ -161,9 +169,73 @@ class PatientSeries:
     def __init__(self, value: Value):
         self.value = value
 
+    @staticmethod
+    def _get_other(other: Expression) -> bool | str | int | Value:
+        if isinstance(other, PatientSeries):
+            return other.value
+        return other
+
+    def __gt__(self, other: Expression) -> Comparator:
+        return self.value > self._get_other(other)
```

Comment:
I had expected that we would be returning a new `PatientSeries` here:

```
    def __gt__(self, other: Expression) -> PatientSeries:
        return type(self)(self.value > self._get_other(other))
```

This would allow us to do eg `(age >= 18) & (age < 25)` entirely with DSL objects, so that users never see QM objects.

Reply:
Oh! I missed that. Yes, I'm sure that's right. The ultimate query model shouldn't have support for syntactic niceties like operator overloading (and should be invisible to the study author).